icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/format/PluralRulesTest.java

```diff
@@ -1523,7 +1542,8 @@ private void generateLOCALE_SNAPSHOT() {
             System.out.println("\n        // " + set);
             for (PluralRules rule : rules) {
                 Set<ULocale> locales = data.get(rule);
-                System.out.print("        \"" + CollectionUtilities.join(locales, ","));
+                String toShow = locales.stream().map(ULocale::toString).collect(Collectors.joining(","));
```

Comment:
Just FYI -- I wonder how many people have to write this kind of work around because the various Java 8 joiner APIs want CharSequences instead of Objects...

Reply:
I don't know.
I thought about that.
My guess is that if it is too easy then people will pass objects without noticing and will get junk (because many objects don't have a friendly `toString`). In fact even the `Locale.toString()` is not that friendly (`zh_TW_#Hant` :-)

Probably a nice signature would be to take a type the collection on `T`, separators, prefix / suffix (optional) AND a method / lambda that maps from T to `String`.

Should be a few lines of code.
And then you can do something like `String.join(locales, Locale::toLanguageTag, ", ")`